TZC380 autoconfiguration (search correct region size) and support for i.MX6Q/D

Emantor opened the pull request:
Hello,

this PR adds a new function called `tzc380_auto_configure` which searches for the matching configuration for a given start address and size. This configuration is then applied with the given permissions to the controller.
This vastly simplifies configurations for devices with a generic RAM layout, since the necessary information is already available from the header file.
Support for i.MX6QD devices is added as a user of the new function.

Emantor commented on core/arch/arm/plat-imx/drivers/tzasc.c:
```diff
@@ -0,0 +1,45 @@
+#include <drivers/tzc380.h>
+#include <imx-regs.h>
+#include <imx.h>
+#include <initcall.h>
```
unused include

Emantor commented on core/arch/arm/plat-imx/drivers/tzasc.c:
```diff
@@ -0,0 +1,45 @@
+ * Rouven Czerwinski <entwicklung@pengutronix.de>
+ */
+
+#include <assert.h>
```
unused include

etienne-lms commented on core/arch/arm/plat-imx/drivers/tzasc.c:
```diff
@@ -0,0 +1,45 @@
+#if defined(CFG_MX6Q) || defined(CFG_MX6D)
+void imx_configure_tzasc(void)
+{
+
```
remove empty line

etienne-lms commented on core/arch/arm/plat-imx/imx6.c:
```diff
@@ -36,6 +36,7 @@
 #include <mm/core_memprot.h>
 #include <mm/core_mmu.h>
 #include <platform_config.h>
+#include <imx.h>
```
suggest to move this up.

Emantor commented:
@etienne-lms Thanks for the review, I addressed your comments and applied the provided tags.
(I just noticed that separate review commits are more common here, I however already rebased and applied the fixups, sorry.)

Emantor commented:
@MrVan I need your Ack for at least 6e3f9164ff3dce1e3e0fbb7aa9d2cde4ac2fa414
@jforissier, @jenswi-linaro any comments on this PR?

jforissier commented on core/arch/arm/plat-imx/imx6.c:
```diff
@@ -77,5 +78,8 @@ void plat_cpu_reset_late(void)
 		     addr != CSU_BASE + CSU_CSL_END;
 		     addr += 4)
 			io_setbits32(addr, CSU_SETTING_LOCK);
+#if defined(CFG_TZC380)
```
Conditional not needed (assuming `imx.h` is changed as suggested above)

jforissier commented:
For "plat-imx: mx6: add support for the TZC380 to MX6Q": LGTM in principle.
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

jforissier commented:
Sorry pasted the wrong commit subject above, now fixed.

MrVan commented:
Acked-by: Peng Fan <peng.fan@nxp.com>

Emantor commented:
@mrvan for the whole series or for the plat-imx commit?

jforissier commented on core/arch/arm/plat-imx/imx.h:
```diff
@@ -33,5 +33,10 @@ uint32_t imx_soc_type(void);
 void imx_gpcv2_set_core_pgc(bool enable, uint32_t offset);
+
+#if !(defined(CFG_MX6Q) || defined(CFG_MX6D) || defined(CFG_TZC380))
+void imx_configure_tzasc(void) {}
```
`static inline void` please.

Minor: it's usually written the other way around, without a negation (if defined() ... else ... endif)

Emantor replied:
I also forgot the `#` for `else`. Fixed with a new Review commit and corrected logic.

etienne-lms commented on core/arch/arm/plat-imx/imx.h:
```diff
@@ -34,9 +34,9 @@ void imx_gpcv2_set_core1_pdn_by_software(void);
-#if !(defined(CFG_MX6Q) || defined(CFG_MX6D) || defined(CFG_TZC380))
-void imx_configure_tzasc(void) {}
-else
+#if (defined(CFG_MX6Q) || defined(CFG_MX6D)) && defined(CFG_TZC380)
```
`#if defined(CFG_TZC380)` should be enough.

etienne-lms commented on core/drivers/tzc380.c:
```diff
@@ -216,6 +216,68 @@ uint32_t tzc_get_action(void)
+
+		/* Case region fits alignment and covers requested area */
+		if ((address % region_size == 0) &&
+			((address + lsize) % region_size == 0)) {
```
please address this minor comment.

Emantor replied:
I missed this, should be good now 👍

Emantor commented:
Squashed down the commits and applied the tags.

jbech-linaro commented:
Thanks @Emantor , patch merged!

Emantor commented:
Thanks for the reviews @etienne-lms @jforissier
